Mathlib/MeasureTheory/Integral/IntegrableOn.lean

```diff
@@ -695,3 +695,24 @@ theorem integrableOn_Iic_iff_integrableOn_Iio :
   integrableOn_Iic_iff_integrableOn_Iio' (by rw [measure_singleton]; exact ENNReal.zero_ne_top)
 
 end PartialOrder
+
+namespace MeasureTheory
+variable {X : Type*} [MeasurableSpace X] {μ : Measure X} {s : Set X}
+theorem IntegrableOn.iff_ofReal {f : X → ℝ} :
+    IntegrableOn f s μ ↔ IntegrableOn (fun x ↦ (f x : ℂ)) s μ :=
+    MeasureTheory.Integrable.iff_ofReal
+
+theorem IntegrableOn.ofReal {f : X → ℝ} (hf : IntegrableOn f s μ) :
+    IntegrableOn (fun x => (f x : ℂ)) s μ := IntegrableOn.iff_ofReal.1 hf
+
+theorem IntegrableOn.re_im_iff {f : X → ℂ} :
+    IntegrableOn (fun x => (f x).re) s μ ∧ IntegrableOn (fun x => (f x).im) s μ ↔
+    IntegrableOn f s μ := Integrable.re_im_iff (f := f)
+
+theorem IntegrableOn.re {f : X → ℂ} (hf : IntegrableOn f s μ) :
+    IntegrableOn (fun x => (f x).re) s μ  := (IntegrableOn.re_im_iff.2 hf).left
+
+theorem IntegrableOn.im {f : X → ℂ} (hf : IntegrableOn f s μ) :
+    IntegrableOn (fun x => (f x).im) s μ := (IntegrableOn.re_im_iff.2 hf).right
+
+end MeasureTheory
```

Comment:
I'm sorry it wasn't clear, when I said "copy the RCLike section of `L1Space`, I literally meant it, in particular writing those with a function taking values in an RCLike field rather than complex numbers.

Reply:
Ohh I see. Okay, will fix this

Reply:
fixed